src/unity/toolkits/activity_classification/activity_classifier.cpp

```diff
@@ -440,23 +446,70 @@ void activity_classifier::perform_training_iteration() {
     float batch_loss = std::accumulate(loss_batch.data(),
                                        loss_batch.data() + loss_batch.size(),
                                        0.f, std::plus<float>());
+
+    
     cumulative_batch_loss += batch_loss / batch.batch_info.size();
 
+    
+    const shared_float_array& output = results.at("output");
+    float cumulative_per_batch_accuracy = 0.f;
+    
+    for (size_t i = 0; i < batch.batch_info.size(); ++i) {
+      
+      shared_float_array output_chunk = output[i];
+
+      // Actual labels 
+      shared_float_array classes_chunk = batch.labels[i];
+      data_iterator::batch::chunk_info info = batch.batch_info[i];
+      
+
+      const float* output_ptr = output_chunk.data();
+      const float* truth_ptr = classes_chunk.data();
+      
+      size_t cumulative_samples = 0;
+      size_t seq_len =0;
+      float cumulative_seq_accuracy = 0.f;
+
+
+      while (cumulative_samples < info.num_samples) {
+
+        // Copy the probability vector to get accuracy.
+        std::copy(output_ptr, output_ptr + num_classes, preds.begin());
+        output_ptr += num_classes;
+        
+        cumulative_samples += prediction_window;
+        seq_len++;
+        truth_ptr ++;
```

Comment:
Hmm, I believe you need to increment `truth_ptr` after you use it in line 483. Note that as-is, you're never looking at the first ground-truth label.

In this case, it might help to follow the idiom of incrementing all the loop variables at the end of the loop body